components/browser/engine-gecko-nightly/src/main/java/mozilla/components/browser/engine/gecko/GeckoEngineSession.kt

```diff
@@ -152,6 +152,13 @@ class GeckoEngineSession(
         geckoSession.goForward()
     }
 
+    /**
+     * See [EngineSession.goToHistoryIndex]
+     */
+    override fun goToHistoryIndex(index: Int) {
+        geckoSession.gotoHistoryIndex(index)
```

Comment:
What happens if you provide an index outside the range of the index available? We should check to see what the different engine requirements and see if we can follow the same pattern for them all.

Maybe something like, [`coerceIn`](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.ranges/coerce-in.html) either at the engine-level or at the use case level?

Reply:
`GeckoSession` seems to ignore invalid values. I'm not sure how to test `SystemEngineSession` though

Reply:
@person808 in android components you can switch the build variant to use the System engine with the Sample Browser.

<img width="495" alt="Screen Shot 2020-07-09 at 6 38 43 PM" src="https://user-images.githubusercontent.com/1370580/87097533-71584980-c213-11ea-9f85-91aee9bb4fe6.png">

Reply:
`SystemEngineSession` also ignores invalid values. Maybe it would be best to document that implementations should ignore invalid values? Coercing to a valid value would mean accessing `BrowserStore` to get the history for the session since AFAIK thats the only way to get the session history for `GeckoSession`. What do you think?

Reply:
Documenting sounds good. This is a particularly awkward API that uses an index value, no need to make it more complex.